web/src/components/ChatMessages/ChatMessageComponent.tsx

```diff
@@ -9,27 +9,38 @@ import type { MessagesContext } from "./types";
 
 type ChatMessageProps = Pick<
   MessagesContext,
-  "deleteMessage" | "updateMessage"
+  "deleteMessage" | "updateMessage" | "availableRoles"
 > & { message: ChatMessageWithId };
 
 export const ChatMessageComponent: React.FC<ChatMessageProps> = ({
   message,
   updateMessage,
   deleteMessage,
+  availableRoles,
 }) => {
   const textAreaRef = useRef<HTMLTextAreaElement | null>(null);
   const [textAreaRows, setTextAreaRows] = useState(1);
+  const [roleIndex, setRoleIndex] = useState(1);
 
   const toggleRole = () => {
     if (message.role === ChatMessageRole.System) return;
 
-    updateMessage(
-      message.id,
-      "role",
-      message.role === ChatMessageRole.User
-        ? ChatMessageRole.Assistant
-        : ChatMessageRole.User,
-    );
+    if (!!availableRoles) {
```

Comment:
*suggestion*: This existence check might return true for empty array of availableRoles which might not be intended